src/runtime/pkg/resourcecontrol/utils_linux.go

```diff
@@ -21,39 +21,15 @@ import (
 // DefaultResourceControllerID runtime-determined location in the cgroups hierarchy.
 const DefaultResourceControllerID = "/vc"
 
-// ValidCgroupPathV1 returns a valid cgroup path for cgroup v1.
+// ValidCgroupPath returns a valid cgroup path.
 // see https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#cgroups-path
-func ValidCgroupPathV1(path string, systemdCgroup bool) (string, error) {
+func ValidCgroupPath(path string, v2 bool) (string, error) {
```

Comment:
Looks like the logic was changed due to `systemdCgroup` removal

- before: the function **throws an error** if `systemdCgroup == true` and the path is not in the format of systemd cgroup.
- now: giving a new path without any errors.

Reply:
@justxuewei Thank you for your review, this issue is indeed worthy of discussion.

The IsSystemdCgroup() is repeated. When calling the ValidCgroupPath() method, the incoming systemdCgroup is obtained through IsSystemdCgroup(path).
This logic seems to be intended to handle the following situation: when calling the `ValidCgroupPath()`, the parameter `systemdCgroup==true` is actively passed in, but in fact, `IsSystemdCgroup(path)` is judged to be `false` inside the function.

I'm not sure if I want to keep this design from before, please give advice, thanks.

Reply:
I still think they are different. Let us list all possible cases here.

| index | path        | systemdCgroup | result             |
|-------|-------------|---------------|--------------------|
| 1     | systemd     | true          | (origin path, nil) |
| 2     | systemd     | false         | (origin path, nil) |
| 3     | non-systemd | true          | ("", error)        |
| 4     | non-systemd | false         | (new path, nil)    |

Case 3 means I expect the path is in the format of systemd, but it isn't actually. Therefore, an error should be thrown. The case is unreachable if `systemdCgroup` is removed. Wdyt?

Reply:
You are right

Reply:
I fixed this question and the email issue.